lib/modules/datasource/docker/index.ts

```diff
@@ -824,42 +824,113 @@ export class DockerDatasource extends Datasource {
     },
   })
   override async getDigest(
-    { registryUrl, packageName }: GetReleasesConfig,
+    { registryUrl, packageName, currentValue, currentDigest }: DigestConfig,
     newValue?: string
   ): Promise<string | null> {
     const { registryHost, dockerRepository } = getRegistryRepository(
       packageName,
       registryUrl!
     );
     logger.debug(
-      `getDigest(${registryHost}, ${dockerRepository}, ${newValue})`
+      `getDigest(${registryHost}, ${dockerRepository}, ${
+        newValue ?? 'undefined'
+      })`
     );
     const newTag = newValue ?? 'latest';
     let digest: string | null = null;
     try {
-      let manifestResponse = await this.getManifestResponse(
-        registryHost,
-        dockerRepository,
-        newTag,
-        'head'
-      );
-      if (manifestResponse) {
-        if (hasKey('docker-content-digest', manifestResponse.headers)) {
+      let architecture: string | null = null;
+      if (currentValue && currentDigest) {
+        const manifestResponse = await this.getManifestResponse(
+          registryHost,
+          dockerRepository,
+          currentValue
+        );
+        if (manifestResponse) {
+          const manifestList = JSON.parse(manifestResponse.body) as
+            | ImageList
+            | Image
+            | OciImageList
+            | OciImage;
+          if (
+            manifestList.schemaVersion === 2 &&
+            manifestList.mediaType === MediaType.manifestListV2
```

Comment:
```suggestion
            (manifestList.mediaType === MediaType.manifestListV2 || manifestList.mediaType === MediaType.ociManifestIndexV1)
```

Reply:
needs further checks, see 
https://github.com/renovatebot/renovate/blob/2d5c8230d0c8b6faf19bf947ddc46336d50e5618/lib/modules/datasource/docker/index.ts#L513-L517

Reply:
Added all checks equivalent to `getConfigDigest()` now